### PR TITLE
Adding ROCm based docker file

### PR DESCRIPTION
**Pull request by jerryyin**
Left out the rocm-prof related changes on purpose. Will open a new PR right after this one merged, just so that I get another chance to test this out. The corresponding jenkins job is under `MLIR/rocm-docker`. I have set up a separate push hook to trigger the job from github. The implication here is that CI test green check-mark does not guarantee the correctness of the dockerfile. It is only after push that the dockerfile's correctness can be verified.

In the future, we should make sure:
 - When there is a dockerfile change, the owner **must** test it out locally to make sure its correctness.
 - When there is a dockerfile change, dockerfile change should be the one and the only one change in that PR.

### CODE REVIEW

**Comment by jerryyin** [comment]
jenkins: retest this please

**Comment by jerryyin** [comment]
jenkins: retest this please

**Comment by jerryyin** [comment]
jenkins: retest this please

File: mlir/utils/jenkins/Dockerfile.rocm [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# This Dockerfile provides a the enviornment for a ROCm installation of MLIR
```
Comment (yiqian1) [inline mlir/utils/jenkins/Dockerfile.rocm]: Remove 'a'.

**Review by yiqian1 — APPROVED** [review]

**Pull request merged**